### ISSUE TITLE
Does not add provided dependencies to classpath

### ISSUE THREAD

**Issue by randomcoder** [sksamuel/sbt-scapegoat#34]
If a project contains some dependencies that are `provided` rather than in `compile` or `test` scope then they are not added to the `scapegoat:managedClasspath` resulting in compile failures.

`project/Dependencies.scala` (snippet)

```
// Scalactic utilities
  val scalactic = "org.scalactic" %% "scalactic" % scalatestVersion % "provided"

  val jodaTime = "joda-time" % "joda-time" % jodaTimeVersion
  val jodaConvert = "org.joda" % "joda-convert" % jodaConvertVersion
```

Output of `show scapegoat:managedClasspath`

```
[info] utils/scapegoat:managedClasspath
[info]  List(Attributed(/home/tim/.ivy2/cache/org.scala-lang/scala-library/jars/scala-library-2.11.7.jar), Attributed(/home/tim/.ivy2/cache/com.sksamuel.scapegoat/scalac-scapegoat-plugin_2.11/jars/scalac-scapegoat-plugin_2.11-1.0.0.jar), Attributed(/home/tim/.ivy2/cache/org.scala-lang/scala-reflect/jars/scala-reflect-2.11.6.jar), Attributed(/home/tim/.ivy2/cache/org.scala-lang.modules/scala-xml_2.11/bundles/scala-xml_2.11-1.0.2.jar), Attributed(/home/tim/.ivy2/cache/org.scaldi/scaldi_2.11/jars/scaldi_2.11-0.4.jar), Attributed(/home/tim/.ivy2/cache/joda-time/joda-time/jars/joda-time-2.8.1.jar), Attributed(/home/tim/.ivy2/cache/org.joda/joda-convert/jars/joda-convert-1.7.jar), Attributed(/home/tim/.ivy2/cache/org.clapper/grizzled-slf4j_2.11/jars/grizzled-slf4j_2.11-1.0.2.jar), Attributed(/home/tim/.ivy2/cache/org.slf4j/slf4j-api/jars/slf4j-api-1.7.7.jar), Attributed(/home/tim/.ivy2/cache/ch.qos.logback/logback-classic/jars/logback-classic-1.1.3.jar), Attributed(/home/tim/.ivy2/cache/ch.qos.logback/logback-core/jars/logback-core-1.1.3.jar), Attributed(/home/tim/.ivy2/cache/org.codehaus.groovy/groovy/jars/groovy-2.4.3.jar))
[info] root/scapegoat:managedClasspath
[info]  List(Attributed(/home/tim/.ivy2/cache/org.scala-lang/scala-library/jars/scala-library-2.11.7.jar), Attributed(/home/tim/.ivy2/cache/com.sksamuel.scapegoat/scalac-scapegoat-plugin_2.11/jars/scalac-scapegoat-plugin_2.11-1.0.0.jar), Attributed(/home/tim/.ivy2/cache/org.scala-lang/scala-reflect/jars/scala-reflect-2.11.6.jar), Attributed(/home/tim/.ivy2/cache/org.scala-lang.modules/scala-xml_2.11/bundles/scala-xml_2.11-1.0.2.jar), Attributed(/home/tim/.ivy2/cache/org.scaldi/scaldi_2.11/jars/scaldi_2.11-0.4.jar))
```

Adding the (provided) dependency to the Scapegoat `libraryDependencies` also doe snot work:

```
libraryDependencies in Scapegoat ++= Seq(scalactic)
```

**Comment by JoshRosen** [comment]
+1. This is a blocker to using this plugin in many of my projects, including https://github.com/databricks/spark-redshift (and most third-party Spark packages, for that matter, since they all declare Spark as a provided dependency).

**Comment by JoshRosen** [comment]
@randomcoder @sksamuel, I came up with a tiny patch which fixes this issue: #37. It might not be the _best_ fix, but it works.
